scripts/src/app/application.ts

```diff
@@ -1,217 +1,77 @@
-/// <reference path="./storage.ts" />
-/// <reference path="../core/configuration.ts" />
-/// <reference path="../core/dom.ts" />
-/// <reference path="../components/types.ts" />
-/// <reference path="../components/registry.ts" />
-/// <reference path="../view/main.tsx" />
-
-"use strict";
-
-module fugazi.app {
-	export const version = {
-		code: "1.0.16",
-		name: "The Argument",
-		toString: function() {
-			return `${ this.name } (${ this.code })`;
-		}
-	};
-
-	export const Events = {
-		Loaded: "app.loaded",
-		Ready: "app.ready"
-	};
-
-	var context: ApplicationContext,
-		mainView: view.MainView,
-		CONTEXT_ID_PARAMS: utils.GenerateIdParameters = {
-			min: 5,
-			max: 10,
-			prefix: "context:"
-		};
-
-	export type Variable = {
-		name: string,
-		type: components.types.Type,
-		value: any
+import * as viewMain from "../view/main";
+import * as utils from "../core/utils";
+import * as registry from "../components/registry";
+import * as converters from "../components/converters";
+import * as types from "../components/types";
+
+
+var context: ApplicationContext;
+export function getContext() {
+	return context;
+}
+export function setContext(newContext: ApplicationContext) {
+	context = newContext;
+}
+
+var mainView: viewMain.MainView;
```

Comment:
done.

there are a lot of `var`s roaming free in the code. some of them should be `const` and some `let`.

phase 2 IMO.